x-pack/plugin/index-lifecycle/src/main/java/org/elasticsearch/xpack/indexlifecycle/PolicyStepsRegistry.java

```diff
@@ -33,18 +34,14 @@
     private Map<String, Step> firstStepMap;
     // keeps track of a mapping from policy/step-name to respective Step
     private Map<String, Map<Step.StepKey, Step>> stepMap;
-
-    public PolicyStepsRegistry() {
-        this.lifecyclePolicyMap = new TreeMap<>();
-        this.firstStepMap = new HashMap<>();
-        this.stepMap = new HashMap<>();
-    }
```

Comment:
I think it might be cleaner to maintain this constructor and have the StepsFactory passed into it. Then it can call `this(new TreeMap<>(), new HashMap<>(), new HashMap<>(), stepsFactory);` which means the caller doesn't have to worry about what kinds of maps are best for the way it uses them internally

Reply:
agreed, I was 50/50 on this at the time of deletion